### PR TITLE
update the analysis options for tools/licenses

### PR DESCRIPTION
**Pull request by devoncarew**
- update the analysis options for tools/licenses

This is an update to https://github.com/flutter/engine/pull/27331; we need to do a little bit of work to the repo to reduce the number of analysis options files. This directory was using a different (and I assume much older) analysis options file from the rest of the repo.

cc @zanderso for review (unless there's someone more familiar with this area)

### CODE REVIEW

File: tools/licenses/lib/filesystem.dart [hunk 1]
```diff
@@ -358,6 +358,12 @@ abstract class Directory extends IoNode {
   Iterable<IoNode> get walk;
 }
 
+extension DirectoryList on Directory {
```
Comment (zanderso) [inline tools/licenses/lib/filesystem.dart:361]: Please avoid extension methods: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-extension

**Review by zanderso — COMMENTED** [review]
Thanks for this cleanup!

File: tools/licenses/lib/filesystem.dart (hunk 1 above)
Comment (devoncarew) [reply tools/licenses/lib/filesystem.dart:361]: It's an extension that's just used in this file. I can change the visibility to private and add a comment?

This is to get around a super odd type hierarchy in filesystem.dart.

I could also change this to a static helper method.

Comment (devoncarew) [reply tools/licenses/lib/filesystem.dart:361]: I updated the call sites to use a static helper method.

**Review by zanderso — APPROVED** [review]
lgtm

**Pull request merged**